docs/docs/code/nodejs/managing-state/README.md

```diff
@@ -0,0 +1,132 @@
+---
+short_description: Store and read data with the built in database.
+thumbnail: https://res.cloudinary.com/pipedreamin/image/upload/v1646763735/docs/icons/icons8-database-96_iv1oup.png
+---
+
+# Managing state
+
+In Node.js (Javascript) code steps, you can also store and retrieve data in code steps.
+
+This is very useful for tracking data between runs of a particular workflow.
+
+:::warning
+This functionality (`$.service.db`) is limited to only Node.js code steps at this time.
+
+Other step languages like [Python](/code/python/), [Bash](/code/bash/) and [Go](/code/go/) do not have this feature available yet.
+
+For more information on what functionality is available for those languages, please refer to their documentation.
+:::
+
+### Injecting the database
+
+By default, Node.js steps don't have access to the database service. It needs to be injected by defining it as a `prop`. 
+
+```javascript
+export default defineComponent({
+  props: {
+    // Define that the "db" variable in our component is a database
+    db: "$.service.db",
+  },
+  async run({ steps, $ }) {
+    // Now we can access the database at "this.db"
+    this.db.set("name", "Dylan")
+  }
+});
+```
+
+:::tip
+`props` injects variables under `this` scope in components.
+
+In the above example we essentially instructed that this step needs the database injected into the `this.db` prop. 
```

Comment:
IMO it's a best practice to include the English description above the code, so you could move this up and say something like:

In this example, you'll:

1. Make the database service available to your component
2. Set a test value at a database key using `this.db.set`

Then the user knows what to look out for in the code.